web/client-api/src/main/java/io/deephaven/ide/shared/IdeSession.java

```diff
@@ -66,17 +68,17 @@ public IdeSession(
 
     // TODO: #37: Need SmartKey support for this functionality
     @JsIgnore
-    public Promise<JsTreeTable> getTreeTable(String name) {
-        return connection.getTreeTable(name, result);
+    public Promise<JsTreeTable> getTreeTable(JsVariableDefinition name) {
```

Comment:
Why did all these methods (`getTable`, `getTreeTable`, `getFigure`) all change to require a variable definition now? That's changing the existing API, and doesn't really give us anything of value over just passing the `name` (or `id`) of the variable.